### ISSUE TITLE
Add a way to make get_pkgconfig_variable override prefix + all *dir options

### ISSUE THREAD

**Issue by emersion** [mesonbuild/meson#6874]
A use-case for `get_pkgconfig_variable` is to figure out where files should be installed. For instance Bash and Fish have a `completionsdir` variable. However we never want to install anything outside the prefix, and we want to obey the datadir (and other *dir options) set by the user.

It would be nice to add a way to make `get_pkgconfig_variable` use the project's values for `prefix` and `*dir` instead of using the pkg-config file's.

**Comment by xclaesse** [comment]
You probably have `completiondir=${prefix}/data/something` so you can use `dep.get_pkgconfig_variable('completiondir', define_variable: 'prefix='+get_option('prefix'))`. Does that work in your case, or I misunderstood something?

**Comment by emersion** [comment]
That doesn't take into account `datadir`. For instance the pkg-config might use `datadir=${prefix}/share` and `completionsdir=${datadir}/bash/completions`. If the user sets `-Ddatadir=myownsharedir -Dprefix=/prefix` on the Meson CLI we probably want to install completions to `/prefix/myownsharedir/bash/completions`.

**Comment by xclaesse** [comment]
In that case you can do `define_variable: ['datadir', get_option('prefix') / get_option('datadir')]`.

**Comment by emersion** [comment]
Yes. However this relies on the pkg-config file using `datadir` and not another variable. The point of this issue is to add an easy way to define all the prefix + *dir variables, and not rely on the pkg-config file to use a specific variable.

**Comment by vinipsmaker** [comment]
@emersion would your problem be solved if meson allowed to pass multiple `--define-variable` args? For instance if you were able to:

```
define_variable : {
  'prefix' : get_option('prefix'),
  'datadir' : get_option('prefix') / get_option('datadir'),
}
```

**Comment by emersion** [comment]
No:

> The point of this issue is to add an easy way to define all the prefix + *dir variables, and not rely on the pkg-config file to use a specific variable.

**Comment by lundmar** [comment]
Sadly, this is a common problem with no pretty solution. Not even in meson.
